ci/pkg/ci/ci.go

```diff
@@ -0,0 +1,306 @@
+			for {
+				yamlNode := make(map[string]interface{})
+
+				err = decoder.Decode(&yamlNode)
+				if err != nil {
+					if err != io.EOF {
+						return err
+					}
+					break
+
+				}
+				kind := yamlNode["kind"].(string)
+				if kind == "list" {
+					nodes := yamlNode["items"].([]interface{})
+					for _, node := range nodes {
+						images = append(images, processYamlNode(node.(map[string]interface{}))...)
+					}
+				} else {
+					images = append(images, processYamlNode(yamlNode)...)
+				}
+
+			}
+
+		}
+		return nil
+	})
+	return images, err
+}
+
+func processYamlNode(yamlNode map[string]interface{}) []models.Image {
+	owner := models.Resource{
+		Kind: yamlNode["kind"].(string),
+		Name: yamlNode["metadata"].(map[string]interface{})["name"].(string),
+	}
+	podSpec := GetPodSpec(yamlNode)
+	images := getImages(podSpec.(map[string]interface{}))
+	return funk.Map(images, func(s string) models.Image {
+		return models.Image{
+			Name:  s,
+			Owner: owner,
+		}
+	}).([]models.Image)
+}
+
+var podSpecFields = []string{"jobTemplate", "spec", "template"}
+var containerSpecFields = []string{"containers", "initContainers"}
+
+// GetPodSpec looks inside arbitrary YAML for a PodSpec
+func GetPodSpec(yaml map[string]interface{}) interface{} {
+	for _, child := range podSpecFields {
+		if childYaml, ok := yaml[child]; ok {
+			return GetPodSpec(childYaml.(map[string]interface{}))
+		}
+	}
+	return yaml
+}
+
+func getImages(podSpec map[string]interface{}) []string {
+	images := make([]string, 0)
+	for _, field := range containerSpecFields {
+		containerField, ok := podSpec[field]
+		if !ok {
+			continue
+		}
+		containers := containerField.([]interface{})
+		for _, container := range containers {
+			images = append(images, container.(map[string]interface{})["image"].(string))
+		}
+	}
+	return images
+}
+
+// GetRepoTags returns the repotags from a tarball of a an image.
+func GetRepoTags(path string) ([]string, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	tarReader := tar.NewReader(f)
+	for {
+		header, err := tarReader.Next()
+
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return nil, err
+		}
+		if header.Name != "manifest.json" {
+			continue
+		}
+		bytes, err := ioutil.ReadAll(tarReader)
+		if err != nil {
+			return nil, err
+		}
+		jsonBody := make([]interface{}, 0)
+		err = json.Unmarshal(bytes, &jsonBody)
+		if err != nil {
+			return nil, err
+		}
+		allRepoTags := make([]string, 0)
+		for _, imageDef := range jsonBody {
+			repoTags := imageDef.(map[string]interface{})["RepoTags"].([]interface{})
+			for _, tag := range repoTags {
+				allRepoTags = append(allRepoTags, tag.(string))
+			}
+		}
+		return allRepoTags, nil
+	}
+	return nil, nil
+}
+
+// SendResults sends the results to Insights
+func SendResults(trivyResults []byte, trivyVersion string, polarisVersion string, configurationObject Configuration, token string) error {
```

Comment:
We'll probably want to do something that scales as we add more reports - maybe an array of `Result` structs?

I'm guessing most tools will output to a file, so maybe it makes sense to just write out the trivy results so that we can have a consistent method for accessing results?